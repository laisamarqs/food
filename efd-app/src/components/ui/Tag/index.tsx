import { TagContainer } from './styles'

export type TagProps = {
  text: string
}

const Tag = ({ text }: TagProps) => (
  <TagContainer role="tag-container">{text}</TagContainer>
)

export default Tag