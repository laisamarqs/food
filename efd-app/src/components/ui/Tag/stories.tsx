import type { Meta } from '@storybook/react'

import Tag, { TagProps } from '.'

const meta = {
  title: 'ui/Tag',
  component: Tag,
  parameters: {
    layout: 'centered'
  },
  tags: ['autodocs'],
  args: {
    text: 'Destaque da semana'
  }
} satisfies Meta<typeof Tag>

// eslint-disable-next-line react-refresh/only-export-components
export default meta

export const Template = (args: TagProps) => {
  return <Tag {...args} />
}