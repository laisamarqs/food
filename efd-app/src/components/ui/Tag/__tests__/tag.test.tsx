import '@testing-library/jest-dom'
import { screen } from '@testing-library/dom'
import { render } from '@testing-library/react'

import Tag from '..'
import { expect } from 'vitest'

describe('<Tag />', () => {
  it('should be render with the text "destaque"', () => {
    const { container } = render(<Tag text="destaque" />)

    expect(screen.findByRole('tag-container', { name: /destaque/i }))
    expect(container.firstChild).toMatchSnapshot()
  })
})