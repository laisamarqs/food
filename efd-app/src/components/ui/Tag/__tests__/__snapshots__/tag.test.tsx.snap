// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Tag /> > should be render with the text "destaque" 1`] = `
<div
  class="sc-dcJsrY QRaBo"
  role="tag-container"
>
  destaque
</div>
`;